### ISSUE TITLE
Blank Screen when I try to connect to a aks cluster

### ISSUE THREAD

**Issue by djarami726** [kontena/lens#6940]
**Describe the bug**
Blank Screen when I try to connect to a aks cluster

**To Reproduce**
1. Start openlens, goto clusters
2. Double click on a cluster
3. Blank screen
4. Go back to clusters and it shows as connected.

**Expected behavior**
Should open up the cluster and show all the info about the cluster and also be able to select options from menu

**Screenshots**
![image](https://user-images.githubusercontent.com/107571134/212427315-93763b87-b662-4bab-9e1b-ee9ccc1cb7a3.png)
List of clusters

![image](https://user-images.githubusercontent.com/107571134/212946124-7f1fe512-3c18-4806-abae-798d2349b29f.png)

Blank screen

**Environment (please complete the following information):**
- Lens Version: 6.3.0 and 6.4.0-alpha.2
- OS: [e.g. OSX] Windows 11
- Installation method (e.g. snap or AppImage in Linux): 
downloaded exe from the tags page

**Logs:**
No logs generated on the command line on windows.

**Kubeconfig:**
Quite often the problems are caused by malformed kubeconfig which the application tries to load. Please share your kubeconfig, remember to remove any secret and sensitive information.

I obfuscated with xxxxx the certificate and also the server names.   Problem happens still happens even with one aks config

```
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    server: https://app-xxxxxxxx-dev-aks-1-16ba61c2.hcp.eastus.azmk8s.io:443
  name: app-xxxxxxxx-dev-aks-1
contexts:
- context:
    cluster: app-xxxxxxxx-dev-aks-1
    user: clusterUser_app-xxxxxxxx-dev-eastus_app-xxxxxxxx-dev-aks-1
  name: app-xxxxxxxx-dev-aks-1
current-context: app-xxxxxxxx-dev-aks-1
kind: Config
preferences: {}
users:
- name: clusterUser_app-xxxxxxxx-dev-eastus_app-xxxxxxxx-dev-aks-1
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1beta1
      args:
      - get-token
      - --environment
      - AzurePublicCloud
      - --server-id
      - 6dae42f8-4368-4678-94ff-888888888888
      - --client-id
      - 80faf920-1908-4b52-b5ef-888888888888
      - --tenant-id
      - bc877e61-f6cf-4461-accd-888888888888
      - --login
      - devicecode
      command: kubelogin
      env: null
      provideClusterInfo: false

```

**Additional context**

**Comment by Nokel81** [comment]
Can you post a screenshot of the "blank"-ness?

**Comment by mercer** [comment]
I have the same with 6.3.0 
<img width="1132" alt="image" src="https://user-images.githubusercontent.com/95358/212698603-e2c74897-f598-4d60-96ee-d4adb83df423.png">

**Comment by Nokel81** [comment]
Thanks. Would you be able to share some logs from the developer tools's console?

**Comment by luca147** [comment]
I have the same behaviour, after update to 6.3.0

**Comment by Nokel81** [comment]
6.3.0 contains a known bug which has been fixed. This is more concerning that in 6.4.0-alpha.2 the bug has reappeared

**Comment by smarek** [comment]
I had the same issue, in my case it was permissions problem. My podreader user did not have access to all namespaces. In console "loadAll" failed since I did not have permission to access all namespaces.

Fixed by adding desired/allowed namespaces in cluster settings and reconnecting to it.

**Comment by Nokel81** [comment]
By "access to all namespaces" do you mean "cannot list namespaces"?

**Comment by smarek** [comment]
That as well probably? How can i tell?

**Comment by Nokel81** [comment]
Can you do a `kubectl get namespace`?

**Comment by smarek** [comment]
No, `Error from server (Forbidden): namespaces is forbidden (...) cannot list resource "namespaces" in API GROUP "" at the cluster scope`

**Comment by Nokel81** [comment]
Okay so you should have gotten a notification about not being able to list namespaces. Maybe that should also include information that it will prevent lens from showing some resources.

**Comment by smarek** [comment]
I can confirm, with version 6.3.0, i get no notification, screen remains blank, the same way as here https://github.com/lensapp/lens/issues/6940#issuecomment-1384129129

**Comment by Nokel81** [comment]
Thanks for the information.

**Comment by noahshogan** [comment]
Facing same issue
Running next steps seems help bring the ui back on

<img width="1368" alt="image" src="https://user-images.githubusercontent.com/26882997/212779230-c133afea-f817-4329-997b-ac156d101be0.png">

**Comment by florimm** [comment]
Also for me worked when I open developers tools and hit F5 then I was able to connect to cluster (see resources)

**Comment by Nokel81** [comment]
There is a bug with 6.3.0 with the blank screen. Does this still happen on 6.4.0-alpha.3?

**Comment by dmckernanacv** [comment]
I am also getting this on Apple silicon, however Windows is fine. I have permissions for namespaces on both instances.

**Comment by Nokel81** [comment]
@dmckernanacv What version are you on?

**Comment by dmckernanacv** [comment]
> @dmckernanacv What version are you on?

6.3.0

**Comment by Nokel81** [comment]
Okay thanks. That has been fixed by #6835. Does it still happen for you in 6.4.0-alpha.3?

**Comment by Nokel81** [comment]
@djarami726 Can you please post a confirmation screenshot of this happening on 6.4.0-alpha.3. If not I will close this issue as a duplicate of #5462

**Comment by djarami726** [comment]
> Can you post a screenshot of the "blank"-ness?

Sorry, I pasted the same one twice.

![image](https://user-images.githubusercontent.com/107571134/212946030-96807453-3f56-4110-a3f7-63c5655cbb95.png)

**Comment by Nokel81** [comment]
If you open up the devtools are there any errors logged in the console?

**Comment by djarami726** [comment]
> 

No more blank screen - however, on one of my clusters, I'm missing several options - everything under workloads and also names spaces. 
![image](https://user-images.githubusercontent.com/107571134/212957126-cd1e3438-56ed-4bc2-add8-1535a4f4fe9f.png)

I've confirmed that I can do a kubectl get pods -A or a kubectl get ns -A.
I do notice that my kubectl get pods -A result is smaller but I do see several namespaces including kube-system.  When I click on the list of Namespaces on the top right, I do get a complete list of my namespaces, but when I select for example - kube-system, no pods show up.

![image](https://user-images.githubusercontent.com/107571134/212957277-a3a93758-7a7f-4537-b7dc-ef82a63cd30f.png)

I also tried on another cluster, and things work fine there.

**Comment by Nokel81** [comment]
Okay thanks for the information. Sounds similar to https://github.com/lensapp/lens/issues/6858

**Comment by dmarjanovic** [comment]
> There is a bug with 6.3.0 with the blank screen. Does this still happen on 6.4.0-alpha.3?

@Nokel81 for me this bug is solved in 6.4.0-alpha.3. Tnx

**Comment by Nokel81** [comment]
@dmarjanovic I think there might be still a bug relating to AKS that we need to track down.

**Comment by djarami726** [comment]
> @dmarjanovic I think there might be still a bug relating to AKS that we need to track down.

@Nokel81 @dmarjanovic Let me know what you need from me to track this issue down.  I want to help.

**Comment by kinyat** [comment]
I have the same issue with EKS as well so it is not only for AKS
Downgraded to v6.2.5 for now

<img width="2554" alt="image" src="https://user-images.githubusercontent.com/1476974/213071109-ddd110a1-18d1-4691-9178-3cee0059acc9.png">

**Comment by tbvl** [comment]
I also have the issue on both 6.3.0 and v6.4.0-alpha.3 on Windows Laptop and AKS. 
![image](https://user-images.githubusercontent.com/1332978/213102961-5c42ac8b-1139-4038-8769-df775df95b4a.png)

Downgrading to 6.2.5 brings it all back:
![image](https://user-images.githubusercontent.com/1332978/213103823-791f3541-8309-43f5-898a-252a65d39db0.png)

**Comment by alexrosenfeld10** [comment]
I get the same problem with IBM GCP and Azure clusters on 6.3.0. The Logs show:

```
[LENS-VIEW]: not initializing view; unknown clusterId="undefined"
__webpack_modules__.1246.BrowserConsole.log @ BrowserConsole.js:41
BrowserConsole.js:41 [LENS-VIEW]: refreshing iframe views, visible cluster id=null
BrowserConsole.js:41 [LENS-VIEW]: refreshing iframe views, visible cluster id=2985e153e8c6b534f06af1f83c8b168d
BrowserConsole.js:41 [LENS-VIEW]: cluster id=2985e153e8c6b534f06af1f83c8b168d should now be visible
BrowserConsole.js:41 [LENS-VIEW]: refreshing iframe views, visible cluster id=undefined
BrowserConsole.js:41 [LENS-VIEW]: not initializing view; unknown clusterId="undefined"
__webpack_modules__.1246.BrowserConsole.log @ BrowserConsole.js:41
BrowserConsole.js:41 [LENS-VIEW]: refreshing iframe views, visible cluster id=null
BrowserConsole.js:41 [LENS-VIEW]: refreshing iframe views, visible cluster id=2985e153e8c6b534f06af1f83c8b168d
BrowserConsole.js:41 [LENS-VIEW]: cluster id=2985e153e8c6b534f06af1f83c8b168d should now be visible
```

but despite this, nothing is visible.

If I refresh using command +R , I get a page, but with no data:

<img width="1591" alt="Screen Shot 2023-01-18 at 10 33 31 AM" src="https://user-images.githubusercontent.com/12940966/213214538-d65aab61-f24f-4b11-8e61-aeca04f6e0ce.png">

**Comment by schedin** [comment]
I don't get the blank page when connecting to a K3S+K3D cluster, nor do I get a blank page when connecting to a "pure" K3S cluster.

I get a blank page when connecting to a VMware Tanzu Kubernetes Grid cluster. However for me the workaround to press Ctrl + R to reload works! It gives me a page with full data (at least what I can tell). I use OpenLens 6.3.0.

Edit: Perhaps not all things work. I cannot open a shell nor see logs on Pods.

**Comment by alexrosenfeld10** [comment]
> Edit: Perhaps not all things work. I cannot open a shell nor see logs on Pods.

@schedin this is not a bug.... see https://github.com/lensapp/lens/issues/1690 . You can add it back yourself using https://github.com/alebcay/openlens-node-pod-menu

**Comment by djarami726** [comment]
> > Edit: Perhaps not all things work. I cannot open a shell nor see logs on Pods.
> 
> @schedin this is not a bug.... see #1690 . You can add it back yourself using https://github.com/alebcay/openlens-node-pod-menu

@alexrosenfeld10 Thank you so much - I was wondering where these very useful options went to.  I tried it (terminal & logs) and it works nice.

**Comment by djarami726** [comment]
> @Nokel81 @dmarjanovic

@Nokel81 @dmarjanovic  Using I wrote a script that refreshes/re-does the `az aks get-credentials` for all the clusters and then I use  6.4.0-alpha.3 and things are running pretty good now.

**Comment by Nokel81** [comment]
@djarami726 Interesting, I guess AKS doesn't support `SelfSubjectRulesReview` without that?

**Comment by djarami726** [comment]
> `SelfSubjectRulesReview`

@Nokel81 Sounds like it, but I don't know enough it.  I could take some logs when it fails and share them.

**Comment by Nokel81** [comment]
Looking through https://github.com/lensapp/lens/blob/master/src/common/cluster/request-namespace-list-permissions.injectable.ts the only reason that something isn't shown is that if a `SelfSubjectRulesReview` succeeds to be created but comes back with no `apiGroups` or `resources`.

**Comment by Nokel81** [comment]
This has now been fixed by the recent patch releases to 6.4
